### PR TITLE
Moving bluetooth module of mijia crate out to new bluez-async crate.

### PR DESCRIPTION
**Pull request by qwandor**
As requested in #91. I changed my mind.

### CODE REVIEW

**Comment by alsuren** [comment]
I really don't want to do this lightly. 

- [ ] We should at least talk to other rust bluetooth maintainers.

I've asked the btleplug maintainer (who also contributed the web-bluetooth bindings in web-sys) for their thoughts on API design. https://www.reddit.com/r/rust/comments/klvd4o/buttplugrs_hits_v1_milestone/ghdrgrg?utm_source=share&utm_medium=web2x&context=3 .

(I've not read your code yet. I will do the other PRs first)

**Comment by qwandor** [comment]
I've commented on https://github.com/deviceplug/btleplug/issues/69 suggesting that I could implement a btleplug backend based on this crate. From having a look at the `btleplug` code, I think it wouldn't be too hard to do. I'd probably make a wrapper struct following the `btleplug` API but async, and then implement the API trait with a bunch of methods just blocking on the async versions. This would make it easy when `btleplug` moves to actually exposing an async API in future.

`dbus-tokio` just merged a PR to move to `tokio` 1.0, so I should be able to upgrade us as well once that makes it to a release. I'll just need to figure out how to use 0.2, 0.3 and 1.0 together the same binary (unless our other dependencies also upgrade before then).

For scope, my intention is to support (tokio-based) async, GAP/GATT central functionality (i.e. acting as a peripheral is out of scope), via BlueZ. I've added support for device-specific streams in #114.

File: bluez-async/Cargo.toml
Comment (qwandor) [inline bluez-async/Cargo.toml:7]: Explain the scope.

**Comment by qwandor** [comment]
> * [x]  Think about scope
>   
>   * [x]  We currently only care about GATT properties, and we only care about reading/subscribing to an event stream at a global level. We also only care about being a client. If this is as far as we want to go, how do we communicate that this is the scope of our project?
Added notes to the README and description.

>   * [x]  if someone starts with our library and finds that they want to escape and do something non-gatt, can we provide them an escape hatch, so that they can use bluez-generated on raw proxy objects? (How do we make this feel sufficiently dirty that people don't do it when they should be using our in-scope APIs?)
They can convert the `DeviceId` or whatever to a dbus `Path`, and then construct a `Proxy` to use as they wish.

>   * [ ]  how are we avoiding the fate of blurz (everything hand coded to use sync dbus calls and would be painful to port to tokio, so the whole thing ended up unmaintained)?
Well, it's using tokio already at least!

>   * [x]  how are we avoiding the fate of bluez-dbus (interfaces for all of the possible concurrency models and never become confident enough to post on crates.io)?
I only intend to support async, and I am confident to post it on crates.io.

File: bluez-async/Cargo.toml
Comment (qwandor) [reply bluez-async/Cargo.toml]: Done.

File: bluez-async/README.md
Comment (alsuren) [inline bluez-async/README.md:13]: I was about to suggest `maintenance = { status = "experimental" }` but then I re-read https://doc.rust-lang.org/cargo/reference/manifest.html#the-badges-section and I guess this is just as good.

**Review by alsuren — APPROVED** [review]
Sorry I blocked this for so long.

**Pull request merged**
